opm/grid/CpGrid.hpp

```diff
@@ -826,8 +828,12 @@ namespace Dune
             //  Add Leaf View to data_.
             (this-> data_).push_back(leaf_view_ptr);
             current_view_data_ = data_[2].get();
+            lgr_names_[lgr_name] = 1;
```

Comment:
Does this means that all LGR's will be at the same leaf view level?

Reply:
Do you need the grid index information in the opm-grid?

Reply:
So far, only one refinement (lgr) is supported via refinePatch() (selection a patch of cells, blocked-shaped, from the global grid, and refine them). We assume the coarse grid to be level 0 (meaning data_[0] of the CpGrid object). The LGR generated via refinement is stored in data_[1] "level 1". The leafview is available in data_[2] but not considered "level 2", since it's the leafview (the cells coming from the coarse grid and that haven't been refined, plus the cells coming from the LGR).

Hope this helps.

Reply:
I'm not sure what grid index means, but I can say that the level index of an LGR (level_ member in a CpGridData object; currently, only 0 [for "GLOBAL"] or 1 [for "LGR", the refinement of a patch of cells from global grid] ) will be needed later to get the correct grid level for an LGR. 

For example, to access the LGR with name "LGR1":

// Find level for LGR
auto level = grid.getLgrNameToLevel().find("LGR1");

// Get corresponding grid view
const auto& lgr = grid.getLevelGridView(level);

Hope this helps.

Reply:
In the commercial simulator each grid has an index. Global is 0, LGR1 would be 1, LGR2 would be 2, and so on...I am thinking if this mapping is done in opm-common, maybe you wouldn't need to know the names here. And for outputting, I believe we may only need grid indexes also.

Reply:
I see, then they seems to represent the same feature (the grid index and what it's call here -- opm-grid -- level index). I can imagine that the map connecting lgr names and integers representing grid/level indices could be useful when the names help the user to distinguish between different lgrs. Example, instead of having "LGR1" and "LGR2", one might prefer to have "LGR_bottom_right_cell", "LGR_left_boundary_corner_cell". 

So far that only one lgr is supported, I agree that it's not very useful. We can also remove it, in case this type of information is not so relevant.

Reply:
Yes, the names itself are not relevant for the code, the grids are numbered in the order they are entered in the input file I think. Outputting the names may be more user-friendly, but I believe we can only visualize the grid indexes in ResInsight (will double check it now to be sure). @bska Do you know what information goes in the binaries? The names may be useful tough for error/warning messages.

Reply:
Well, the information needs to be somewhere. Where is probably not that important as long the simulator is able to
retrieve the correct level for an LGR. That can be done perfectly well even without this PR as long as the LGRs and hence the levels are constructed in the right (or a deterministic) order by the simulator. I guess this would be natural anyway. The parser is probably making that information available in the correct order anyway, right?